app/interactors/create_user.rb

```diff
@@ -0,0 +1,25 @@
+class CreateUser
+  include Interactor
+
+  def call
+    validate_input
+    execute
+    validate_output
+  end
+
+  private
+
+  def validate_input
+    context.fail!(error: "invalid user params") unless context.user_params
```

Comment:
Yeah so here you are just making sure that the context has `user_params`. It wouldn't throw and `ArgumentError` if you don't provide them though. The `.call` method takes a single optional `Hash` parameter, which is then converted into the context. This spec is to make sure that you do provide a `Hash` to the context with `user_params` as a key, but nothing else.

Some examples:

```ruby
CreateUser.call(user: user_params) # fail
CreateUser.call(user_params: {}) # pass
CreateUser.call() # fail
CreateUser.call(param1: 1, param2: 2) # fail
```

This validation level is good. I've user interactors like this. But you can also have a more strict validation. Should `user_params` be a `Hash` and if so what keys is it required to have? :dollar: